### ISSUE TITLE
writing / reading nippy looses column metadata

### ISSUE THREAD

**Issue by behrica** [techascent/tech.ml.dataset#331]
```clojure
(def df
  (->
   (ds/->dataset {})
   (ds/add-column (ds-col/new-column :a [0] {:k :v}))))

(-> df :a meta)
;; => {:k :v, :name :a, :datatype :int64, :n-elems 1}

(ds/write! df "/tmp/out.nippy")

(-> (ds/->dataset "/tmp/out.nippy") :a meta)
;; => {:name :a, :datatype :int64, :n-elems 1}

```

{:k :v} is missing after read

**Comment by cnuernber** [comment]
That is no good.  It should definitely work.

**Comment by behrica** [comment]
The root cause of it is here:
https://github.com/techascent/tech.ml.dataset/blob/7a6f9c616598f32c73aea5a08f6a9ec11d271e62/src/tech/v3/dataset/base.clj#L1024

In that line any metadata except ":name" and ":datatype" is ignored.

**Comment by behrica** [comment]
fixed by #332
